lib/location/method_gnss.c

```diff
@@ -0,0 +1,347 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdio.h>
+#include <zephyr.h>
+#include <logging/log.h>
+
+#include <modem/location.h>
+#include <modem/lte_lc.h>
+#include <modem/at_cmd.h>
+#include <nrf_modem_gnss.h>
+#include <nrf_errno.h>
+#if defined(CONFIG_NRF_CLOUD_AGPS)
+#include <net/nrf_cloud_rest.h>
+#include <net/nrf_cloud_agps.h>
+#endif
+
+#include "loc_core.h"
+
+LOG_MODULE_DECLARE(location, CONFIG_LOCATION_LOG_LEVEL);
+
+/* range 10240-3456000000 ms, see AT command %XMODEMSLEEP */
+#define MIN_SLEEP_DURATION_FOR_STARTING_GNSS 10240
+#define AT_MDM_SLEEP_NOTIF_START "AT%%XMODEMSLEEP=1,%d,%d"
+#define AT_MDM_SLEEP_NOTIF_STOP "AT%XMODEMSLEEP=0" /* not used at the moment */
+#define AGPS_REQUEST_RECV_BUF_SIZE 3500
+#define AGPS_REQUEST_HTTPS_RESP_HEADER_SIZE 500
```

Comment:
We need to fine tune these values. Currently the size of the full A-GPS data seems to be 3005 bytes, but can it change? So some wiggle room might be appropriate if we allocate memory statically. Also I have no idea about the potential maximum size of the HTTPS response header, seems to work with 500 bytes for the time being.

Reply:
@hiltunent Please check if this is still valid and do necessary actions